### PR TITLE
Fix issue when during a release management wrong plan was picked up

### PR DESCRIPTION
**Pull request by AlexeiVainshtein**

### CODE REVIEW

File: src/main/java/org/jfrog/bamboo/release/vcs/AbstractVcsManager.java [hunk 1]
```diff
@@ -44,10 +45,14 @@ protected BuildContext getContext() {
...
-        for (int i = 0; i < tasks.size(); i++) {
-            if (tasks.get(i) != null &&
-                    tasks.get(i).getPluginKey().contains("bamboo-artifactory-plugin")) {
-                return tasks.get(i).getConfiguration();
+        for (TaskDefinition taskDefinition : tasks) {
+            // Check that task definition enabled.
+            if (taskDefinition.isEnabled()) {
+                AbstractBuildContext config = AbstractBuildContext.createContextFromMap(taskDefinition.getConfiguration());
+                // Check the release management is enabled
+                if ((config != null) && config.releaseManagementContext.isReleaseMgmtEnabled()) {
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/bamboo/release/vcs/AbstractVcsManager.java]: Redundant parentheses around ```(config != null)```.

File: src/main/java/org/jfrog/bamboo/util/TaskDefinitionHelper.java [hunk 2]
```diff
@@ -82,9 +81,13 @@ public static TaskDefinition findMavenDefinition(List<? extends TaskDefinition>
...
     public static TaskDefinition findReleaseTaskDefinition(List<? extends TaskDefinition> taskDefinitions) {
         if (taskDefinitions != null) {
-            for (TaskDefinition definition : taskDefinitions) {
-                if ("true".equals(definition.getConfiguration().get(ENABLE_RELEASE_MANAGEMENT))) {
-                    return definition;
+            for (TaskDefinition taskDefinition : taskDefinitions) {
+                if (taskDefinition.isEnabled()) {
+                    AbstractBuildContext config = AbstractBuildContext.createContextFromMap(taskDefinition.getConfiguration());
+                    // Check the release management is enabled
+                    if ((config != null) && config.releaseManagementContext.isReleaseMgmtEnabled()) {
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/bamboo/util/TaskDefinitionHelper.java]: Redundant parentheses around ```(config != null)```.
Also, this method is also identical to the ```getTaskConfiguration``` method. I suggest reusing the code.

File: src/main/java/org/jfrog/bamboo/release/vcs/AbstractVcsManager.java (hunk 1 above)
Comment (AlexeiVainshtein) [reply src/main/java/org/jfrog/bamboo/release/vcs/AbstractVcsManager.java]: Per our discussion, the config can be null so this check is not redundant.

File: src/main/java/org/jfrog/bamboo/util/TaskDefinitionHelper.java (hunk 2 above)
Comment (AlexeiVainshtein) [reply src/main/java/org/jfrog/bamboo/util/TaskDefinitionHelper.java]: Per our discussion, the config can be null so this check is not redundant. I did a reuse for the code.

**Pull request merged**